### PR TITLE
update Hydra-maester docu to reflect recent changes + rbac fix

### PR DESCRIPTION
**Pull request by jakkab**
## Related issue

https://github.com/ory/hydra-maester/pull/19

## Proposed changes

update Hydra-maester docu to reflect changes implemented in https://github.com/ory/hydra-maester/pull/19

## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md)
- [x] I have read the [security policy](../security/policy)
- [x] I confirm that this pull request does not address a security vulnerability. If this pull request addresses a security
vulnerability, I confirm that I got green light (please contact [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push the changes.
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation within the code base (if appropriate)

### CODE REVIEW

File: docs/helm/hydra-maester.md [hunk 1]
```diff
@@ -23,7 +23,57 @@ The syntax of the CR **Spec** fields is a simplified representation of the the O
...
+### Update OAuth2 client password
+ 
+To change the password of your OAuth2 client, follow these steps:
+
+1. Save the ID of your client and the updated password to a new Kubernetes Secret.
+2. Edit the instance of `oauth2clients.hydra.ory.sh/v1alpha1` CR by replacing the value of the `SecretName` property with the name of the new Secret.
+
+## Troubleshooting
+
+If the controller encounters en error while processing an instance of `oauth2clients.hydra.ory.sh/v1alpha1` CR, it adds the **status** section which describes the cause of the error. This table lists the fields of the **status** section.
```
Comment (tomekpapiernik) [inline docs/helm/hydra-maester.md]:
```suggestion
If the controller encounters en error while processing an instance of the `oauth2clients.hydra.ory.sh/v1alpha1` CR, it adds the **status** section which describes the cause of the error. This table lists the fields of the **status** section.
```

File: docs/helm/hydra-maester.md [hunk 2]
```diff
@@ -23,7 +23,57 @@ The syntax of the CR **Spec** fields is a simplified representation of the the O
...
+## Credentials
+
+Apart from interacting with ORY Hydra's API, Hydra Maester controls Kubernetes Secrets that store credentials of the registered clients. You can use those credentials to interact with corresponding OAuth2 clients registered by ORY Hydra. For each client, you can either specify the client ID and secret or let Hydra OAuth2 server generate a random client ID and secret.
+
+### Use your own credentials
+
+To create a client with your own credentials, follow these steps:
+
+1. Save your client ID and secret to a Kubernetes Secret.
+2. Create an instance of `oauth2clients.hydra.ory.sh/v1alpha1` CR with the `SecretName` property set to the name of the Secret that contains the credentials.
```
Comment (tomekpapiernik) [inline docs/helm/hydra-maester.md]:
```suggestion
2. Create an instance of the `oauth2clients.hydra.ory.sh/v1alpha1` CR with the **SecretName** property set to the name of the Secret that contains the credentials.
```

File: docs/helm/hydra-maester.md [hunk 3]
```diff
@@ -23,7 +23,57 @@ The syntax of the CR **Spec** fields is a simplified representation of the the O
...
+
+> **NOTE:** Both resources must exist in the same namespace.
```
Comment (tomekpapiernik) [inline docs/helm/hydra-maester.md]:
```suggestion
> **NOTE:** Both resources must exist in the same Namespace. 
```

File: docs/helm/hydra-maester.md [hunk 4]
```diff
@@ -23,7 +23,57 @@ The syntax of the CR **Spec** fields is a simplified representation of the the O
...
+
+1. Save your client ID and secret to a Kubernetes Secret.
+2. Create an instance of `oauth2clients.hydra.ory.sh/v1alpha1` CR with the `SecretName` property set to the name of the Secret that contains the credentials.
+
+> **NOTE:** Both resources must exist in the same namespace. 
+
+
+### Use Hydra-generated credentials
+
+If you don't need specific client ID and password, use Hydra-generated credentials. Create an instance of `oauth2clients.hydra.ory.sh/v1alpha1` CR with the `SecretName` property set to a string of your choice. The property is used by Hydra-maester to create a Kubernetes Secret that will store the credentials of the registered client.
```
Comment (tomekpapiernik) [inline docs/helm/hydra-maester.md]:
```suggestion
If you don't need a specific client ID and password, use Hydra-generated credentials. Create an instance of the`oauth2clients.hydra.ory.sh/v1alpha1` CR with the **SecretName** property set to a string of your choice. The property is used by Hydra Maester to create a Kubernetes Secret that will store the credentials of the registered client.
```

File: docs/helm/hydra-maester.md [hunk 5]
```diff
@@ -23,7 +23,57 @@ The syntax of the CR **Spec** fields is a simplified representation of the the O
...
+
+### Update the OAuth2 client secret
+ 
+To change the password of your OAuth2 client, follow these steps:
+
+1. Save the ID of your client and the updated password to a new Kubernetes Secret.
+2. Edit the instance of `oauth2clients.hydra.ory.sh/v1alpha1` CR by replacing the value of the `SecretName` property with the name of the new Secret.
```
Comment (tomekpapiernik) [inline docs/helm/hydra-maester.md]:
```suggestion
2. Edit the instance of the `oauth2clients.hydra.ory.sh/v1alpha1` CR by replacing the value of the **SecretName** property with the name of the new Secret.
```

File: docs/helm/hydra-maester.md [hunk 6]
```diff
@@ -23,7 +23,57 @@ The syntax of the CR **Spec** fields is a simplified representation of the the O
...
+| **status.reconciliationError.description** | Description of the reconciliation error                            |
+
+
+### Status codes
+
+These are the status codes used to describe reconciliation errors:
+
+| Code                           |  Description                                                      |
+|--------------------------------|-------------------------------------------------------------------|
+| **CLIENT_REGISTRATION_FAILED** | unable to register an OAuth2Client.                               |
```
Comment (tomekpapiernik) [inline docs/helm/hydra-maester.md]:
```suggestion
| **CLIENT_REGISTRATION_FAILED** | Unable to register an OAuth2Client                               |
```

File: docs/helm/hydra-maester.md [hunk 7]
```diff
@@ -23,7 +23,57 @@ The syntax of the CR **Spec** fields is a simplified representation of the the O
...
+| **SECRET_CREATION_FAILED**     | unable to create a secret with the name specified in the CR Spec. |
```
Comment (tomekpapiernik) [inline docs/helm/hydra-maester.md]:
```suggestion
| **SECRET_CREATION_FAILED**     | Unable to create a secret with the name specified in the CR Spec |
```

File: docs/helm/hydra-maester.md [hunk 8]
```diff
@@ -23,7 +23,57 @@ The syntax of the CR **Spec** fields is a simplified representation of the the O
...
+| **CLIENT_UPDATE_FAILED**       | unable to update an existing OAuth2Client.                        |
```
Comment (tomekpapiernik) [inline docs/helm/hydra-maester.md]:
```suggestion
| **CLIENT_UPDATE_FAILED**       | Unable to update an existing OAuth2Client                        |
```

File: docs/helm/hydra-maester.md [hunk 9]
```diff
@@ -23,7 +23,57 @@ The syntax of the CR **Spec** fields is a simplified representation of the the O
...
+| **INVALID_SECRET**             | the existing secret specified in the CR Spec is invalid           |
```
Comment (tomekpapiernik) [inline docs/helm/hydra-maester.md]:
```suggestion
| **INVALID_SECRET**             | The existing secret specified in the CR spec. is invalid           |
```

**Comment by aeneasr** [comment]
Thank you! :)

**Pull request merged**